src/core/algorithms/dd/dd_verifier/dd_verifier.cpp

```diff
@@ -0,0 +1,195 @@
+}
+
+void DDVerifier::CheckCorrectnessDd() const {
+    auto check_constraint = [](auto const &constraint) {
+        if (constraint.constraint.upper_bound < constraint.constraint.lower_bound) {
+            throw std::invalid_argument("Invalid constraint bounds for column: " +
+                                        constraint.column_name);
+        }
+        if (constraint.constraint.lower_bound < 0 || constraint.constraint.upper_bound < 0) {
+            throw std::invalid_argument("Constraint bounds cannot be negative for column: " +
+                                        constraint.column_name);
+        }
+    };
+
+    std::ranges::for_each(dd_.left, check_constraint);
+    std::ranges::for_each(dd_.right, check_constraint);
+}
+
+unsigned long long DDVerifier::ExecuteInternal() {
+    auto elapsed_milliseconds = util::TimedInvoke(&DDVerifier::VerifyDD, this);
+    PrintStatistics();
+    return elapsed_milliseconds;
+}
+
+void DDVerifier::CheckDFOnRhs(std::vector<std::pair<std::size_t, std::size_t>> const &lhs) {
+    for (auto const &pair : lhs) {
+        auto curr_constraint = dd_.right.cbegin();
+        bool is_error = false;
+        for (auto const column_index : rhs_column_indices_) {
+            if (double const dif = CalculateDistance(column_index, pair);
+                !curr_constraint->constraint.Contains(dif)) {
+                highlights_.emplace_back(column_index, pair, dif);
+                is_error = true;
+            }
+            ++curr_constraint;
+        }
+        if (is_error) {
+            ++num_error_rhs_;
+        }
+    }
+}
+
+void DDVerifier::VerifyDD() {
+    auto get_column_index = [&](auto const &constraint) {
+        return typed_relation_->GetSchema()->GetColumn(constraint.column_name)->GetIndex();
+    };
+    std::ranges::transform(dd_.left, std::back_inserter(lhs_column_indices_), get_column_index);
+    std::ranges::transform(dd_.right, std::back_inserter(rhs_column_indices_), get_column_index);
+    for (auto const col : lhs_column_indices_) {
+        if (!IsColumnMetrizable(col)) {
+            std::string message;
+            message += std::to_string(col);
+            message += " column is not metrizable";
+            throw std::invalid_argument(message);
+        }
+    }
+    for (auto const col : rhs_column_indices_) {
+        if (!IsColumnMetrizable(col)) {
+            std::string message;
+            message += std::to_string(col);
+            message += " column is not metrizable";
+            throw std::invalid_argument(message);
+        }
+    }
+    CheckCorrectnessDd();
+    num_rows_ = typed_relation_->GetNumRows();
+    num_columns_ = typed_relation_->GetNumColumns();
+
+    std::vector<std::pair<std::size_t, std::size_t>> const lhs = GetRowsWhereLhsHolds();
+    CheckDFOnRhs(lhs);
+    error_ = lhs.empty() ? 0 : static_cast<double>(num_error_rhs_) / lhs.size();
+}
+
+std::vector<Highlight> const &DDVerifier::GetHighlights() const {
+    return highlights_;
+}
```

Comment:
is that what clang-format formats like? I thought it supposed to do ` const& `. Strange